auth/api/iam/jar.go

```diff
@@ -92,14 +105,14 @@ func createJarRequest(client did.DID, authServerURL string, modifier requestObje
 	}
 	return jarRequest{
 		Claims:           oauthParams,
-		Client:           client,
+		Client:           clientID,
 		RequestURIMethod: requestURIMethod,
 	}
 }
 
 func (j jar) Sign(ctx context.Context, claims oauthParameters) (string, error) {
-	clientID := claims.get(oauth.ClientIDParam)
-	clientDID, err := did.ParseDID(clientID)
+	issuerID := claims.get(jwt.IssuerKey)
```

Comment:
future refactor we'll introduce a `Client` that has an `ID` property and `SigningKey()` function..